src/renderer/pages/Staking/Operations/Bond/Bond.tsx

```diff
@@ -146,73 +144,34 @@ export const Bond = () => {
     const transactions = getBondTxs(Object.keys(validators));
 
     if (signer && isMultisig(txAccounts[0])) {
-      const multisigTx = getMultisigTx(txAccounts[0], signer.accountId, transactions[0]);
-      setMultisigTx(multisigTx);
+      const wrapAsMulti = {
+        signatoryId: signer.accountId,
+        account: txAccounts[0],
+      };
+      setWrappers([wrapAsMulti]);
     }
 
-    setTransactions(transactions);
+    setTxs(transactions);
     setValidators(validators);
     setActiveStep(Step.CONFIRMATION);
   };
 
   const getBondTxs = (validators: Address[]): Transaction[] => {
     return txAccounts.map(({ accountId }) => {
       const address = toAddress(accountId, { prefix: addressPrefix });
-      const commonPayload = { chainId, address };
 
-      const bondTx = {
-        ...commonPayload,
-        type: TransactionType.BOND,
-        args: {
-          value: stakeAmount,
-          controller: address,
-          payee: destination?.type === RewardsDestination.TRANSFERABLE ? { Account: destination.address } : 'Staked',
-        },
-      };
+      const bondTx = buildTransaction(TransactionType.BOND, address, chainId, {
```

Comment:
Continuing discussion of ExtrinsicBuilder started in the ExtrinsicService, we can take this call as an example.

Instead of passing the parameters to the Extrinsic Service and create a huge switch case there we can apply inversion of control. That is being, instead buildTransaction must accept a closure with ExtrinsicBuilder parameter and each module like Bond will use extrinsic builder to set call and parameters.

This will allow calls logic incapsulation inside module they are being used in. If we need such transaction logic shared between modules then we can put it in a sharing function. The second point is that we get rid of switch...case in the ExtrinsicService and make it transaction agnostic which allows him concentrate only on general transaction parameters such as mortality, genesis hash and so on.